datalad_neuroimaging/extractors/fslfeat.py

```diff
@@ -0,0 +1,276 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""FSL Feat analysis configuration and result metadata extractor"""
+from __future__ import absolute_import
+
+from os.path import join as opj
+from os.path import exists
+from os.path import basename
+from os.path import dirname
+from os.path import relpath
+import re
+import logging
+lgr = logging.getLogger('datalad.metadata.extractors.fslfeat')
+from datalad.log import log_progress
+from datalad.metadata.definitions import vocabulary_id
+from datalad.metadata.extractors.base import BaseMetadataExtractor
+
+
+context = {
+    'fslfeat': {
+        # no term definitions exists for FEAT's variable names
+        # going with a disambiguating URL for now
+        '@id': 'https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FEAT/',
+        'description': 'non-vocabulary for FSL FEAT variables',
+        'type': vocabulary_id}
+}
+
+_contrastvar_expr = re.compile(r'([^\d]+)([\d]+)\.([\d]+)')
+
+_procedural_switches = {
+    'tsplot_yn', 'poststats_yn', 'melodic_yn', 'stats_yn',
+    'regstandard_yn', 'regstandard_nonlinear_yn', 'filtering_yn',
+}
+
+_ignor_vars = {
+    'fmri(relative_yn)', 'fmri(help_yn)', 'fmri(featwatcher_yn)',
+    'fmri(sscleanup_yn)', 'fmri(zdisplay)', 'fmri(zmin)',
+    'fmri(zmax)', 'fmri(rendertype)', 'fmri(bgimage)',
+    'fmri(constcol)',
+}
+
+_trancode = {
+    'analysis': {
+        '0': 'nofirstlevel',
+        '7': 'full',
+        '1': 'preprocessing',
+        '2': 'statistics',
+    },
+    'level': {
+        '1': 'first',
+        '2': 'higher',
+    },
+    'inputtype': {
+        '1': 'featdir',
+        '2': 'cope',
+    },
+    'mc': {
+        '0': 'none',
+        '1': 'mcflirt',
+    },
+    'st': {
+        '0': 'none',
+        '1': 'regularup',
+        '2': 'regulardown',
+        '3': 'sliceordercfgfile',
+        '4': 'slicetimingcfgfile',
+        '5': 'interleaved',
+    },
+    'motionevs': {
+        '0': False,
+        '1': True,
+    },
+    'mixed_yn': {
+        '0': 'me_ols',
+        '1': 'me_flame1+2',
+        '2': 'me_flame1',
+        '3': 'fixedeffects',
+    },
+    'thresh': {
+        '0': 'none',
+        '1': 'uncorrected',
+        '2': 'voxel',
+        '3': 'cluster',
+    },
+    'tempfilt_yn': {
+        '0': False,
+        '1': True,
+    },
+    'shape': {
+        '0': 'square',
+        '1': 'sinusoid',
+        '2': 'custom_ev1',
+        '3': 'custom_ev3',
+        '4': 'interaction',
+        '10': 'empty',
+    },
+}
+
+
+def _convert_dtype(val):
```

Comment:
Why not to rely on having `""` around the value to immediately decide it to be a string, this way we would avoid converting some "degenerate" evtitle's (especially the ones with explicit leading 0s for some reason such as "001", or "inf" or "NaN" for floats) etc  into numeric values, where they were intended to stay strings.

Reply:
Because an FSF file has strings that are not quoted.